stm32-modules/heater-shaker/include/heater-shaker/heater_task.hpp

```diff
@@ -82,72 +137,129 @@ requires MessageQueue<QueueImpl<Message>, Message> class HeaterTask {
     }
 
     auto visit_message(const messages::SetTemperatureMessage& msg) -> void {
-        auto response =
-            messages::AcknowledgePrevious{.responding_to_id = msg.id};
-        static_cast<void>(task_registry->comms->get_message_queue().try_send(
-            messages::HostCommsMessage(response)));
+        if (state.system_status == State::ERROR) {
+            // While in error state, we will refuse to set temperatures
+            auto response = messages::AcknowledgePrevious{
+                .responding_to_id = msg.id,
+                .with_error = most_relevant_error()};
+            static_cast<void>(
+                task_registry->comms->get_message_queue().try_send(
+                    messages::HostCommsMessage(response)));
+
+        } else {
+            auto response =
+                messages::AcknowledgePrevious{.responding_to_id = msg.id};
+            static_cast<void>(
+                task_registry->comms->get_message_queue().try_send(
+                    messages::HostCommsMessage(response)));
+        }
     }
 
     auto visit_message(const messages::GetTemperatureMessage& msg) -> void {
+        errors::ErrorCode code = pad_a.error != errors::ErrorCode::NO_ERROR
```

Comment:
I am having trouble understanding the logic on this line.

Reply:
Honestly I probably shouldn't do this with a ternary. The idea is to get the error held by either of pad a or pad b, and to end up with no error only if neither has an error code set.